web-gateway/app/views/editItem.scala.html

```diff
@@ -1,14 +1,30 @@
 @import helper._
+@import java.util.Optional
 @import java.time.temporal.ChronoUnit
 @import controllers.{Currency => Currencies}
 
-@(itemForm: Form[ItemForm])(implicit nav: Nav)
+@import com.example.auction.item.api.UpdateItemResultCodes
+@import com.example.auction.item.api.UpdateItemResultCodes._
 
-@main(message("createItem")) {
+@(itemId: UUID, itemForm: Form[ItemForm], errorMessage: Optional[UpdateItemResultCodes] )(implicit nav: Nav)
 
-    <h2>@message("createItem")</h2>
 
-    @foundationForm(itemForm, routes.ItemController.createItem()) {
+@main(message("editItem")) {
+
+    <h2>@message("editItem")</h2>
+
+    @foundationForm(itemForm, routes.ItemController.editItem(itemId.toString)) {
+
+        @if(errorMessage.isPresent) {
```

Comment:
Definitely necessary to handle errors from the server, but I think it ought to also check the current state of the item when the form is rendered and disable fields that shouldn't be editable.